src/OneDriveSdk.WindowsForms/Authentication/AuthenticationContextWrapper.cs

```diff
@@ -106,5 +136,111 @@ public IAuthenticationResult AcquireToken(string resource, string clientId, Uri
+        /// <param name="clientCredential">The client credential of the application.</param>
+        /// <param name="resource">The resource to authenticate against.</param>
+        /// <returns>The <see cref="IAuthenticationResult"/>.</returns>
+        public async Task<IAuthenticationResult> AcquireTokenByAuthorizationCodeAsync(
+            string code,
+            Uri redirectUri,
+            ClientCredential clientCredential,
+            string resource)
+        {
+            var result = await this.authenticationContext.AcquireTokenByAuthorizationCodeAsync(
+                code,
+                redirectUri,
+                clientCredential,
+                resource);
+
+            return result == null ? null : new AuthenticationResultWrapper(result);
+        }
+
+        /// <summary>
+        /// Authenticates the user silently using <see cref="AuthenticationContext.AcquireTokenByAuthorizationCodeAsync(string, Uri, ClientAssertionCertificate, string)"/>.
+        /// </summary>
+        /// <param name="code">The authorization code.</param>
+        /// <param name="redirectUri">The redirect URI for the application.</param>
+        /// <param name="clientAssertionCertificate">The client assertion certificate of the application.</param>
+        /// <param name="resource">The resource to authenticate against.</param>
+        /// <returns>The <see cref="IAuthenticationResult"/>.</returns>
+        public async Task<IAuthenticationResult> AcquireTokenByAuthorizationCodeAsync(
+            string code,
+            Uri redirectUri,
+            ClientAssertionCertificate clientAssertionCertificate,
+            string resource)
+        {
+            var result = await this.authenticationContext.AcquireTokenByAuthorizationCodeAsync(
+                code,
+                redirectUri,
+                clientAssertionCertificate,
+                resource);
+
+            return result == null ? null : new AuthenticationResultWrapper(result);
+        }
+
+        /// <summary>
+        /// Authenticates the user silently using <see cref="AuthenticationContext.AcquireTokenByRefreshToken(string, string, string)"/>.
+        /// </summary>
+        /// <param name="refreshToken">The refresh token.</param>
+        /// <param name="clientId">The client ID for the application.</param>
+        /// <param name="resource">The resource to authenticate against.</param>
+        /// <returns>The <see cref="IAuthenticationResult"/>.</returns>
+        public async Task<IAuthenticationResult> AcquireTokenByRefreshTokenAsync(
+            string refreshToken,
+            string clientId,
+            string resource)
+        {
+            var result = await this.authenticationContext.AcquireTokenByRefreshTokenAsync(
+                refreshToken,
+                clientId,
+                resource);
+
+            return result == null ? null : new AuthenticationResultWrapper(result);
```

Comment:
Hmm, I don't like that duplicated check.  AuthenticationResultWrapper looks like it can handle null values being wrapped?

---
In reply to: [64650372](https://github.com/OneDrive/onedrive-sdk-csharp/pull/72#discussion_r64650372) [](ancestors = 64650372)